File: lib/result/ProfileParameterResult.java
package com.teamlab.communication.wireless.lpr9201.result;

/**
 * リザルト ProfileParameter
 */
public class ProfileParameterResult extends Result {
	/**
	 * 
	 */
	public static final int RESULT_CODE = 0x85;

	/**
	 * 
	 */
	public static final int DATA_LENGTH_SIZE = 1; // byte
	
	protected long value;

	public ProfileParameterResult(int[] datas) {
		super(RESULT_CODE, DATA_LENGTH_SIZE, datas);

		/*
		for (int d : datas) {
			System.out.print(String.format("0x%02X ", d));
		}
		System.out.println();
		*/

		// FIXME 必要なデータ長がない場合の処理を書く
		//this.paramNo = datas[0];

		for (int i = 0; i < datas.length; i++) {
			this.value |= datas[i] << (8 * (datas.length - i - 1));
		}
	}

	public long getValue() {
		return this.value;
	}
}

File: lib/result/profile_parameter_result.js
/* jslint node: true */
"use strict"

/**
 * リザルト ProfileParameter
 */

exports.ProfileParameterResult = ProfileParameterResult;

/**
 * 結果コード
 */
ProfileParameterResult.RESULT_CODE = 0x85;

/**
 * データ長バイトサイズ
 */
ProfileParameterResult.DATA_LENGTH_BYTE_SIZE = 1; // byte

/**
 * @param {Result} result
 */
function ProfileParameterResult(result) {
  this.result = result;

  if (result.datas.length < 8) {
    for (var i = 0, il = result.datas.length; i < il; i++) {
      this.value |= result.datas[i] << (8 * (result.datas.length - i - 1));
    }

    this.valueHigh = 0;
    this.valueLow = this.value;
  } else {
    // high
    for (var i = 0, il = 4; i < il; i++) {
      this.valueHigh |= result.datas[i] << (8 * (result.datas.length - i - 1));
    }

    // low
    for (var i = 4, il = result.datas.length; i < il; i++) {
      this.valueLow |= result.datas[i] << (8 * (result.datas.length - i - 1));
    }
  }
}

/**
 * パース可能か
 * @param {Result} result
 */
ProfileParameterResult.canParse = function(result) {
  return result !== null && result.resultCode === ProfileParameterResult.RESULT_CODE;
}
